src/review/format.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { readFileSync, writeFileSync } from 'fs';
+
+/**
+ * SIZE is a big-endian uint64
+ *
+ * SIZE path
+ * SIZE filecontent
+ */
+
+export const HS = 6;
+export const ENCODING = 'utf8';
+
+export interface Entry {
+  key: string;
+  data: Buffer;
+}
+
+interface Item {
+  buf: Buffer;
+  bytesRead: number;
+}
+
+export function read(path: string): Entry[] {
+  const buf = readFileSync(path);
+  return parseAll(buf);
+}
+
+export function write(path: string, entries: Entry[]): void {
+  const buf = Buffer.allocUnsafe(encodedLength(entries));
+  encodeAll(buf, entries);
+  writeFileSync(path, buf);
+}
+
+export function parseAll(buf: Buffer, loc = 0): Entry[] {
```

Comment:
Here and `encodeAll`, is there any reason for users to pass in a non-zero value for `loc`? If not, maybe it doesn't need to be there?